src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1577,57 +1577,56 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Sum")>]
         let inline sum (source: seq< (^a) >) : ^a = 
-            use e = source.GetEnumerator() 
             let mutable acc = LanguagePrimitives.GenericZero< (^a) >
-            while e.MoveNext() do
-                acc <- Checked.(+) acc e.Current
+            for item in source do
+                acc <- Checked.(+) acc item
```

Comment:
@buybackoff  In any case, it would be great if you could update this PR to only include the first improvement to Seq.average so we can action that.  The rest needs further investigation/numbers/IL-analysis etc. thanks